File: mi.web.form.html/MiEditForm.versão original/versoes_anteriores/versao.30-ok/temp.js
// Definição do tipo de campo (TField) como um objeto em JavaScript
class TField {
    constructor(name = '', value = '', typ = '', size = 0) {
        this.name = name;    // Nome do campo
        this.value = value;  // Valor do campo
        this.typ = typ;      // Tipo do campo
        this.size = size;    // Tamanho do campo
    }
}

class MyClass {
    constructor() {
        // Atributo que contém os dados da propriedade CurrentField
        this.currentField = new TField();

        // Atributo que salva o currentField anterior ao atual
        // Necessário caso se deseje fazer alguma verificação antes de uma ação
        this._currentFieldOld = new TField();
    }

    // Método que define o currentField
    setCurrentField(aCurrentField) {
        this.currentField = aCurrentField;
    }

    // Propriedade que obtém ou define o campo atual
    get currentField() {
        return this.currentField;
    }

    set currentField(aCurrentField) {
        this.setCurrentField(aCurrentField);
    }
}

File: mi.web.form.html/MiEditForm.versão original/versoes_anteriores/versao .34/temp.pas
Por favor converta esse codigo em pascal para javascript.

        type TField = record                              
                        Name : String;
                        Value : String;
                        typ   : string; 
                        size  : Integer;
                      end;
      {$REGION --> Propriedade CurrentField}
          {: O atributo protected **@name** contem os dados da propriedade CurrentField.}
          protected _CurrentField  : TField;


          {: O atributo **@name** salva o currentefield anterior ao atual.

             - **ATENÇÃO**
               - Necessário caso se deseja fazer alguma crítica com os dados do buffer
                 do corrente campo focado ao pressionar um botão de ação.
          }
          public CurrentField_old : TField;

          {: O Método **@name** Set o currenteField com aCurrentField}
          protected Procedure SetCurrentField(aCurrentField : TField);

          {: A propriedade **@name** contém um ponteiro para o campo selecionado }
          public property CurrentField : TField  read _CurrentField write SetCurrentField;

          {: A atributo **@name** contém um ponteiro para o campo selecionado
             e que FieldNum seja diferente de zero.
          }
          public protected CurrentField_focused  : TField;
      {$ENDREGION --> Propriedade CurrentField}  

      